### ISSUE TITLE
Wasabi sync fails to complete - chunk transfers continue forever

### ISSUE THREAD

**Issue by madcatsu** [ncw/rclone#3255]
#### What is the problem you are having with rclone?
Sync operations to Wasabi are endless for files broken up into chunks. Transfers initially transfer quickly but drop to bps speeds and continue for days until cancelled.

#### What is your rclone version (output from `rclone version`)
1.47.0

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Ubuntu 18.04 LTS 64 bit

####  Which cloud storage system are you using? (eg Google Drive)
Wasabi - this is setup as an encrypted destination

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
`rclone sync /source Wasabi-Encrypt:/destination-bucket/data  --checkers=32 --transfers=4 --s3-upload-concurrency=4 --s3-chunk-size=128M --progress`

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

```
2019/06/12 04:26:02 DEBUG : rclone: Version "v1.47.0" starting with parameters ["rclone" "sync" "/source" "Wasabi-Encrypt:/destination-bucket/data "--checkers=32" "--transfers=4" "--s3-upload-concurrency=4" "--s3-chunk-size=128M" "--s3-upload-cutoff=512M" "-vv" "--log-file" "/home/sysadm/scratch/rclone.log"]
2019/06/12 04:26:02 DEBUG : Using config file from "/home/sysadm/.config/rclone/rclone.conf"
2019/06/12 04:26:18 INFO  : Encrypted drive 'Wasabi-Encrypt:/destination-bucket/data': Waiting for transfers to finish
2019/06/12 04:27:03 INFO  : 
Transferred:   	    2.250G / 5.653 GBytes, 40%, 37.833 MBytes/s, ETA 1m32s
Errors:                 0
Checks:              1465 / 1465, 100%
Transferred:            0 / 2, 0%
Elapsed time:      1m0.8s
Transferring:
 * _REDACTED_file01.dat: 39% /2.871G, 4.488M/s, 6m38s
 * _REDACTED_file02.dat: 40% /2.782G, 8.023M/s, 3m31s

2019/06/12 04:28:03 INFO  : 
Transferred:   	    2.250G / 5.653 GBytes, 40%, 19.057 MBytes/s, ETA 3m2s
Errors:                 0
Checks:              1465 / 1465, 100%
Transferred:            0 / 2, 0%
Elapsed time:      2m0.8s
Transferring:
 * _REDACTED_file01.dat: 39% /2.871G, 95.634k/s, 5h19m5s
 * _REDACTED_file02.dat: 40% /2.782G, 170.971k/s, 2h49m21s

2019/06/12 04:29:03 INFO  : 
Transferred:   	    2.250G / 5.653 GBytes, 40%, 12.736 MBytes/s, ETA 4m33s
Errors:                 0
Checks:              1465 / 1465, 100%
Transferred:            0 / 2, 0%
Elapsed time:      3m0.8s
Transferring:
 * _REDACTED_file01.dat: 39% /2.871G, 1.989k/s, 255h33m42s
 * _REDACTED_file02.dat: 40% /2.782G, 3.558k/s, 135h38m15s

2019/06/12 04:30:03 INFO  : 
Transferred:   	    2.250G / 5.653 GBytes, 40%, 9.564 MBytes/s, ETA 6m4s
Errors:                 0
Checks:              1465 / 1465, 100%
Transferred:            0 / 2, 0%
Elapsed time:      4m0.8s
Transferring:
 * _REDACTED_file01.dat: 39% /2.871G, 42/s, 12280h35m59s
 * _REDACTED_file02.dat: 40% /2.782G, 75/s, 6517h50m30s

2019/06/12 04:31:03 INFO  : 
Transferred:   	    2.250G / 5.653 GBytes, 40%, 7.657 MBytes/s, ETA 7m35s
Errors:                 0
Checks:              1465 / 1465, 100%
Transferred:            0 / 2, 0%
Elapsed time:      5m0.8s
Transferring:
 * _REDACTED_file01.dat: 39% /2.871G, 0/s, 590123h36m58s
 * _REDACTED_file02.dat: 40% /2.782G, 1/s, 313203h57m0s
```

I've also redacted all the checkers DEBUG output

**Comment by ncw** [comment]
Hmm, it looked like something just hung in the transfer.

I would hope rclone's timeouts would kick in at this point but they don't appear to have.

Can you do a run with `--vv --dump headers` and we can see what http requests are outstanding?

**Comment by madcatsu** [comment]
I've tried tweaking a bunch of the s3 options, as well as # of transfers all to no effect. Suffice to say that this issue seems to have appeared only recently and is not limited to these specific files, but generally, anything more than a 1-2 GB in size.

Can I share the log output with you via GDrive or OneDrive? Would rather not paste here as it's thousands of lines long and hard to redact ;)

**Comment by ncw** [comment]
> Can I share the log output with you via GDrive or OneDrive? Would rather not paste here as it's thousands of lines long and hard to redact ;)

Sure.... you can share it with nick@craig-wood.com  on Gdrive or Onedrive (or email it - put a link to the issue in) - thanks

**Comment by madcatsu** [comment]
@ncw Shared the output on Gdrive. Feel free to post any relevant snippets here, just nothing incriminating 😆

**Comment by ncw** [comment]
Thanks for that :-)

It looks like that data transfer just stopped at  around `2019/06/12 12:34:01`

Rclone has timeouts for its http transactions which it should be using.  They are idle timeouts and should kick in at 5 minutes by default but don't seem to...

There are 8 outstanding HTTP requests

```
$ egrep 'RESPONSE|REQUEST' rclone-sync-wasabi.log | cut -d' ' -f8 | sort | uniq -c | sort -rn | egrep -v ' [2468]'
      3 0xc001454600)
      3 0xc0005a6a00)
      3 0xc00011c400)
      1 0xc01686d000)
      1 0xc0162a9800)
      1 0xc00221cb00)
      1 0xc000aa9200)
      1 0xc00085a100)
```

The outstanding requests are all PUT requests like this

    2019/06/12 12:32:14 DEBUG : PUT /bucket/dm8pt9/6h9g3j/67rhpe/6bc6s?partNumber=4&uploadId=gDKwO HTTP/1.1

Uploading parts of the two outstanding files.

I think this is probably some kind of networking problem.

I normally tell people to check their routers at this point - some routers can be overwhelmed with lots of upload traffic.  It could be wasabi blocking the traffic, though that seems unlikely.  Or it could be your OS - check ethernet drivers etc.

Rclone's idle timeouts don't appear to have kicked in which is strange but perhaps indicative of where the problem is.

It would probably be worth trying [the latest beta](https://beta.rclone.org) which is compiled with a later go version too.

**Comment by madcatsu** [comment]
Thanks @ncw ,

I'm not in control of the router as it's a physical box co-located at Hetzner in Germany, though I can say I've not had a problem with sync with these exact same parameters until recently, so I suspect there may be something environmental at Hetzner or Wasabi

Will update this issue once I've tested the beta

**Comment by ncw** [comment]
It might be worth checking the `dmesg` on the box to see if you are getting whining from the ethernet card.

**Comment by madcatsu** [comment]
New log snippet here (the previous one I cancelled which is why you saw it just stop).

The idle timeouts do actually kick in as you can see, I should have mentioned earlier that the other logs were for a cancelled transfer.

Have tested version v1.47.0-126-g44f64917-beta - still seeing the same behavior where transfers just drop to nothing.

```
2019-06-13 15:38:58 ERROR : file001.dat: Failed to copy: MultipartUpload: upload multipart failed
        upload id: D7kEaqitneZr-wNEj43EUAbZVSk_FaGQisYc5RrzBhX55A9Lvq0hSfzE4IYMQrZ1_nNl7_BG6wgKKEyDUQwIUfRw0Tw_v4VDto1fYSqrqqCGS8TdLoiLGHxXI0onDZBK
caused by: RequestError: send request failed
caused by: Put https://s3.us-west-1.wasabisys.com/datavault-bucket/dm8pt9210qd6sgj8io4aplgknsg6el21qf64cuuein6lu0mg46c0/6h9g3j7fv2s4ifp7rc4u97vci4/67rhpeifbt3mabb483585dhvjhgphskji4hjhtl2bl24p5tpab90/6bc6s75m8q9a2bc8opi8uut0fs1da8vsotv5uc8ijtgvaad5brhhiualgradsh3c49qc119qcebbi?partNumber=1&uploadId=D7kEaqitneZr-wNEj43EUAbZVSk_FaGQisYc5RrzBhX55A9Lvq0hSfzE4IYMQrZ1_nNl7_BG6wgKKEyDUQwIUfRw0Tw_v4VDto1fYSqrqqCGS8TdLoiLGHxXI0onDZBK: net/http: timeout awaiting response headers
2019-06-13 16:12:35 ERROR : file002.dat: Failed to copy: MultipartUpload: upload multipart failed
        upload id: VHh1uVRnJzkw_RFJwLqEcwPmuqCbrBu3lutXvtON-A-jX3D-lXt4lK85SPiEC8LcF2zpViesDASrXKTF1mW-efNGQe2adGSBZunUohOjL0IeIiKrq0wDs-my7Z89WAGE
caused by: RequestError: send request failed
caused by: Put https://s3.us-west-1.wasabisys.com/datavault-bucket/dm8pt9210qd6sgj8io4aplgknsg6el21qf64cuuein6lu0mg46c0/6h9g3j7fv2s4ifp7rc4u97vci4/tgjapvuratgpiufkov0202jcdlcn0voe578lqqttl500hhcdnc8g/3tdp6b94a1mvurt46eomvfpi4v7pa0fl0vs6bgukf93pmc5m1lfimgm21riaj7jnlv1s7fn6c2kke?partNumber=1&uploadId=VHh1uVRnJzkw_RFJwLqEcwPmuqCbrBu3lutXvtON-A-jX3D-lXt4lK85SPiEC8LcF2zpViesDASrXKTF1mW-efNGQe2adGSBZunUohOjL0IeIiKrq0wDs-my7Z89WAGE: net/http: timeout awaiting response headers
2019-06-13 16:12:35 ERROR : Encrypted drive 'Wasabi-datavault-Encrypt:/datavault-bucket/store': not deleting files as there were IO errors
2019-06-13 16:12:35 ERROR : Encrypted drive 'Wasabi-datavault-Encrypt:/datavault-bucket/store': not deleting directories as there were IO errors
2019-06-13 16:12:35 ERROR : Attempt 1/3 failed with 3 errors and: MultipartUpload: upload multipart failed
        upload id: VHh1uVRnJzkw_RFJwLqEcwPmuqCbrBu3lutXvtON-A-jX3D-lXt4lK85SPiEC8LcF2zpViesDASrXKTF1mW-efNGQe2adGSBZunUohOjL0IeIiKrq0wDs-my7Z89WAGE
caused by: RequestError: send request failed
caused by: Put https://s3.us-west-1.wasabisys.com/datavault-bucket/dm8pt9210qd6sgj8io4aplgknsg6el21qf64cuuein6lu0mg46c0/6h9g3j7fv2s4ifp7rc4u97vci4/tgjapvuratgpiufkov0202jcdlcn0voe578lqqttl500hhcdnc8g/3tdp6b94a1mvurt46eomvfpi4v7pa0fl0vs6bgukf93pmc5m1lfimgm21riaj7jnlv1s7fn6c2kke?partNumber=1&uploadId=VHh1uVRnJzkw_RFJwLqEcwPmuqCbrBu3lutXvtON-A-jX3D-lXt4lK85SPiEC8LcF2zpViesDASrXKTF1mW-efNGQe2adGSBZunUohOjL0IeIiKrq0wDs-my7Z89WAGE: net/http: timeout awaiting response headers
2019-06-13 19:34:00 ERROR : file001.dat: Failed to copy: MultipartUpload: upload multipart failed
        upload id: L9sHoeKrsYIYwOkvcGdiZ9Z6dgWw1M3sxICE9TRgNtzia-iIXBxietRSGBUN2GFbcbDBDfDAWVV8GsamKLz6H-orJpzI4k4njr1s5BZQhFzdike4WEDIxM3wyawuSDuf
caused by: RequestError: send request failed
caused by: Put https://s3.us-west-1.wasabisys.com/datavault-bucket/dm8pt9210qd6sgj8io4aplgknsg6el21qf64cuuein6lu0mg46c0/6h9g3j7fv2s4ifp7rc4u97vci4/67rhpeifbt3mabb483585dhvjhgphskji4hjhtl2bl24p5tpab90/6bc6s75m8q9a2bc8opi8uut0fs1da8vsotv5uc8ijtgvaad5brhhiualgradsh3c49qc119qcebbi?partNumber=1&uploadId=L9sHoeKrsYIYwOkvcGdiZ9Z6dgWw1M3sxICE9TRgNtzia-iIXBxietRSGBUN2GFbcbDBDfDAWVV8GsamKLz6H-orJpzI4k4njr1s5BZQhFzdike4WEDIxM3wyawuSDuf: net/http: timeout awaiting response headers
2019-06-13 19:48:51 ERROR : file002.dat: Failed to copy: MultipartUpload: upload multipart failed
        upload id: fLoBYvnl2Vi6okf3x1y7eRNelKhAcfmKQ4N845SSkdVog1p6wr3VDT7L_QP4UpiwaeaGhnCjy_F6J1Dj-FXkssgrxFpQZrdq4YpJJBGkh4RZWC-YuAckk07ro7wkK9mh
caused by: RequestError: send request failed
caused by: Put https://s3.us-west-1.wasabisys.com/datavault-bucket/dm8pt9210qd6sgj8io4aplgknsg6el21qf64cuuein6lu0mg46c0/6h9g3j7fv2s4ifp7rc4u97vci4/tgjapvuratgpiufkov0202jcdlcn0voe578lqqttl500hhcdnc8g/3tdp6b94a1mvurt46eomvfpi4v7pa0fl0vs6bgukf93pmc5m1lfimgm21riaj7jnlv1s7fn6c2kke?partNumber=1&uploadId=fLoBYvnl2Vi6okf3x1y7eRNelKhAcfmKQ4N845SSkdVog1p6wr3VDT7L_QP4UpiwaeaGhnCjy_F6J1Dj-FXkssgrxFpQZrdq4YpJJBGkh4RZWC-YuAckk07ro7wkK9mh: net/http: timeout awaiting response headers
2019-06-13 19:48:51 ERROR : Encrypted drive 'Wasabi-datavault-Encrypt:/datavault-bucket/store': not deleting files as there were IO errors
2019-06-13 19:48:51 ERROR : Encrypted drive 'Wasabi-datavault-Encrypt:/datavault-bucket/store': not deleting directories as there were IO errors
2019-06-13 19:48:51 ERROR : Attempt 2/3 failed with 3 errors and: MultipartUpload: upload multipart failed
        upload id: fLoBYvnl2Vi6okf3x1y7eRNelKhAcfmKQ4N845SSkdVog1p6wr3VDT7L_QP4UpiwaeaGhnCjy_F6J1Dj-FXkssgrxFpQZrdq4YpJJBGkh4RZWC-YuAckk07ro7wkK9mh
caused by: RequestError: send request failed
caused by: Put https://s3.us-west-1.wasabisys.com/datavault-bucket/dm8pt9210qd6sgj8io4aplgknsg6el21qf64cuuein6lu0mg46c0/6h9g3j7fv2s4ifp7rc4u97vci4/tgjapvuratgpiufkov0202jcdlcn0voe578lqqttl500hhcdnc8g/3tdp6b94a1mvurt46eomvfpi4v7pa0fl0vs6bgukf93pmc5m1lfimgm21riaj7jnlv1s7fn6c2kke?partNumber=1&uploadId=fLoBYvnl2Vi6okf3x1y7eRNelKhAcfmKQ4N845SSkdVog1p6wr3VDT7L_QP4UpiwaeaGhnCjy_F6J1Dj-FXkssgrxFpQZrdq4YpJJBGkh4RZWC-YuAckk07ro7wkK9mh: net/http: timeout awaiting response headers
2019-06-13 23:18:51 ERROR : file002.dat: Failed to copy: MultipartUpload: upload multipart failed
        upload id: qnYXqAkYDx72bPJIPEtSrL5qCWEzkmoFtT3qgCiHC2To0XzF0O-yu5LMo-BxvFFWKHOYd6968dwFocq1-CnGF_UqRrbxyf8LjyP0ZVtHzKxS3oPKmJ7YUkXfDldnq-Mf
caused by: RequestError: send request failed
caused by: Put https://s3.us-west-1.wasabisys.com/datavault-bucket/dm8pt9210qd6sgj8io4aplgknsg6el21qf64cuuein6lu0mg46c0/6h9g3j7fv2s4ifp7rc4u97vci4/tgjapvuratgpiufkov0202jcdlcn0voe578lqqttl500hhcdnc8g/3tdp6b94a1mvurt46eomvfpi4v7pa0fl0vs6bgukf93pmc5m1lfimgm21riaj7jnlv1s7fn6c2kke?partNumber=1&uploadId=qnYXqAkYDx72bPJIPEtSrL5qCWEzkmoFtT3qgCiHC2To0XzF0O-yu5LMo-BxvFFWKHOYd6968dwFocq1-CnGF_UqRrbxyf8LjyP0ZVtHzKxS3oPKmJ7YUkXfDldnq-Mf: net/http: timeout awaiting response headers
2019-06-14 02:53:37 ERROR : file001.dat: Failed to copy: MultipartUpload: upload multipart failed
        upload id: 3LUAnL50i3ON5byX7IFX8Cx4ZBN66yDwNpHc2kTTtsANSaCX0YTN_pu0H5cyq3oc0VJrPuzR1GzzBFUFT-kdsdi0RRBo3NLIRBOXNzBlJvZaVU47fH5m1uMU2RbywoCc
2019/06/14 02:53:37 Failed to sync: MultipartUpload: upload multipart failed
        upload id: 3LUAnL50i3ON5byX7IFX8Cx4ZBN66yDwNpHc2kTTtsANSaCX0YTN_pu0H5cyq3oc0VJrPuzR1GzzBFUFT-kdsdi0RRBo3NLIRBOXNzBlJvZaVU47fH5m1uMU2RbywoCc
caused by: RequestError: send request failed
caused by: Put https://s3.us-west-1.wasabisys.com/datavault-bucket/dm8pt9210qd6sgj8io4aplgknsg6el21qf64cuuein6lu0mg46c0/6h9g3j7fv2s4ifp7rc4u97vci4/67rhpeifbt3mabb483585dhvjhgphskji4hjhtl2bl24p5tpab90/6bc6s75m8q9a2bc8opi8uut0fs1da8vsotv5uc8ijtgvaad5brhhiualgradsh3c49qc119qcebbi?partNumber=1&uploadId=3LUAnL50i3ON5byX7IFX8Cx4ZBN66yDwNpHc2kTTtsANSaCX0YTN_pu0H5cyq3oc0VJrPuzR1GzzBFUFT-kdsdi0RRBo3NLIRBOXNzBlJvZaVU47fH5m1uMU2RbywoCc: dial tcp: lookup s3.us-west-1.wasabisys.com on 127.0.0.53:53: read udp 127.0.0.1:33914->127.0.0.53:53: i/o timeout
```

**Comment by madcatsu** [comment]
OK, so the beta version actually ended up transferring the outstanding files. Transfer duration was much longer than expected, coming in at 1h almost to the letter but without any errors.

**Comment by ncw** [comment]
> The idle timeouts do actually kick in as you can see, I should have mentioned earlier that the other logs were for a cancelled transfer.

OK that is good to know! How long did the timeouts take to happen?  It should have been either 1 minute or 5 minutes.



> OK, so the beta version actually ended up transferring the outstanding files. Transfer duration was much longer than expected, coming in at 1h almost to the letter but without any errors.

I'm still not convinced that this isn't some networking problem, or problem at wasabi. Rclone seems to be working properly.

**Comment by madcatsu** [comment]
The timeouts took between 3-4 hours in some instances, this means that data was transferring but it looks to be at bytes per second speeds which sometimes burst back up to MB per second speeds. 

I had an `iftop` running for the duration and a 5GB transfer ended up retrying nearly 72GB of data. I don't have output handy sorry but will try dropping another couple of large files in to sync and measure again

**Comment by ncw** [comment]
> The timeouts took between 3-4 hours in some instances, this means that data was transferring but it looks to be at bytes per second speeds which sometimes burst back up to MB per second speeds.

A 3-4 hour timeout is a TCP timeout rather than rclone's idle timeouts.

> I had an `iftop` running for the duration and a 5GB transfer ended up retrying nearly 72GB of data. I don't have output handy sorry but will try dropping another couple of large files in to sync and measure again

Ouch!

**Comment by madcatsu** [comment]
Looks like the latest beta has resolved this issue at least, still getting a couple of retry attempts but much improved

**Comment by ncw** [comment]
> Looks like the latest beta has resolved this issue at least, still getting a couple of retry attempts but much improved

That has now been released as [v1.48](https://rclone.org/downloads/) BTW

I'll close this now then but feel free to re-open if you still have problems.
